Traceback while updating nest in 2021.8b0

### The problem

```
2021-07-28 16:47:35 ERROR (ThreadPoolExecutor-ThreadScheduler_0) [google.cloud.pubsub_v1.subscriber._protocol.streaming_pull_manager] Top-level exception occurred in callback while processing a message
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py", line 84, in _wrap_callback_errors
callback(message)
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/google_nest_subscriber.py", line 77, in callback_wrapper
future.result()
File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 445, in result
return self.__get_result()
File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 390, in __get_result
raise self._exception
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/google_nest_subscriber.py", line 269, in _async_message_callback
await self._device_manager_task.result().async_handle_event(event)
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/device_manager.py", line 47, in async_handle_event
await device.async_handle_event(event_message)
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/device.py", line 137, in async_handle_event
await callback(event_message)
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/device.py", line 104, in handle_event
target()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 464, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 498, in _async_write_ha_state
state = self._stringify_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 470, in _stringify_state
state = self.state
File "/usr/src/homeassistant/homeassistant/components/nest/sensor_sdm.py", line 132, in state
return trait.ambient_humidity_percent
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/device_traits.py", line 111, in ambient_humidity_percent
return cast_assert(float, self._data[AMBIENT_HUMIDITY_PERCENT])
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/typing.py", line 10, in cast_assert
assert isinstance(data, t), f"Expected data with type {t} but was {type(data)}"
AssertionError: Expected data with type <class 'float'> but was <class 'int'>
2021-07-28 16:47:35 ERROR (ThreadPoolExecutor-ThreadScheduler_1) [google.cloud.pubsub_v1.subscriber._protocol.streaming_pull_manager] Top-level exception occurred in callback while processing a message
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py", line 84, in _wrap_callback_errors
callback(message)
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/google_nest_subscriber.py", line 77, in callback_wrapper
future.result()
File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 445, in result
return self.__get_result()
File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 390, in __get_result
raise self._exception
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/google_nest_subscriber.py", line 269, in _async_message_callback
await self._device_manager_task.result().async_handle_event(event)
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/device_manager.py", line 47, in async_handle_event
await device.async_handle_event(event_message)
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/device.py", line 137, in async_handle_event
await callback(event_message)
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/device.py", line 104, in handle_event
target()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 464, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 498, in _async_write_ha_state
state = self._stringify_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 470, in _stringify_state
state = self.state
File "/usr/src/homeassistant/homeassistant/components/nest/sensor_sdm.py", line 132, in state
return trait.ambient_humidity_percent
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/device_traits.py", line 111, in ambient_humidity_percent
return cast_assert(float, self._data[AMBIENT_HUMIDITY_PERCENT])
File "/usr/local/lib/python3.9/site-packages/google_nest_sdm/typing.py", line 10, in cast_assert
```

### What is version of Home Assistant Core has the issue?

2021.8b0

### What was the last working version of Home Assistant Core?

2021.7.4

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

nest

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/nest

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

--- a/homeassistant/components/nest/manifest.json
+++ b/homeassistant/components/nest/manifest.json
@@ -4,7 +4,7 @@
   "config_flow": true,
   "dependencies": ["ffmpeg", "http"],
   "documentation": "https://www.home-assistant.io/integrations/nest",
-  "requirements": ["python-nest==4.1.0", "google-nest-sdm==0.3.4"],
+  "requirements": ["python-nest==4.1.0", "google-nest-sdm==0.3.5"],
   "codeowners": ["@allenporter"],
   "quality_scale": "platinum",
   "dhcp": [

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -707,7 +707,7 @@ google-cloud-pubsub==2.1.0
 google-cloud-texttospeech==0.4.0
 
 # homeassistant.components.nest
-google-nest-sdm==0.3.4
+google-nest-sdm==0.3.5
 
 # homeassistant.components.google_travel_time
 googlemaps==2.5.1

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -404,7 +404,7 @@ google-api-python-client==1.6.4
 google-cloud-pubsub==2.1.0
 
 # homeassistant.components.nest
-google-nest-sdm==0.3.4
+google-nest-sdm==0.3.5
 
 # homeassistant.components.google_travel_time
 googlemaps==2.5.1

